recipe/patches/2667-Fix-improper-include-of-build-directory.patch

```diff
@@ -0,0 +1,42 @@
+--- a/configure.ac
++++ b/configure.ac
+@@ -3012,8 +3012,7 @@ SUBFILING_VFD=no
+ HAVE_MERCURY="no"
+
+ ## Always include subfiling directory so public header files are available
+-CPPFLAGS="$CPPFLAGS -I$ac_abs_confdir/src/H5FDsubfiling"
+-AM_CPPFLAGS="$AM_CPPFLAGS -I$ac_abs_confdir/src/H5FDsubfiling"
++H5_CPPFLAGS="$H5_CPPFLAGS -I$ac_abs_confdir/src/H5FDsubfiling"
+
+ AC_MSG_CHECKING([if the subfiling I/O virtual file driver (VFD) is enabled])
+
+@@ -3061,8 +3060,7 @@ if test "X$SUBFILING_VFD" = "Xyes"; then
+     mercury_dir="$ac_abs_confdir/src/H5FDsubfiling/mercury"
+     mercury_inc="$mercury_dir/src/util"
+
+-    CPPFLAGS="$CPPFLAGS -I$mercury_inc"
+-    AM_CPPFLAGS="$AM_CPPFLAGS -I$mercury_inc"
++    H5_CPPFLAGS="$H5_CPPFLAGS -I$mercury_inc"
+
+     HAVE_STDATOMIC_H="yes"
+     AC_CHECK_HEADERS([stdatomic.h],,[HAVE_STDATOMIC_H="no"])
+--- a/configure
++++ b/configure
```

Comment:
@hmaarrfk I am a bit out of my depth, but I added the corresponding changes to `configure`. Do we think this is enough? I'm not sure an easy way to test that this actually has the desired effect...

Reply:
My test for this kind of stuff is "build locally, and manually check". If it has been approved upstream it builds confidence that we don't need a longer term test